app/components/file-viewer/index.jsx

```diff
@@ -21,39 +23,53 @@ const VIEWERS = {
   image: ImageViewer,
   text: TextViewer,
   video: VideoPlayer,
-  audio: AudioPlayer
+  audio: AudioPlayer,
+  application: CanvasViewer
 };
 
 function subjectViewerSelector(props) {
   if (Array.isArray(props.type)) {
     if (props.type.includes('audio')) {
       return VIEWERS.audio;
     }
-    // ... add outher here if neccessary
+    // ... add other here if neccessary
   }
   return VIEWERS[props.type] || DefaultViewer;
 }
 
 function FileViewer(props) {
   const Viewer = subjectViewerSelector(props);
-
+  const viewerProps = {
+    className: props.className,
+    style: props.style,
+    src: props.src,
+    type: props.type,
+    format: props.format,
+    frame: props.frame,
+    onLoad: props.onLoad,
+    onFocus: props.onFocus,
+    onBlur: props.onBlur
+  };
+  if (props.type === 'application') {
+    Object.assign(
+      viewerProps,
+      {
+        annotation: cloneDeep(props.annotation),
+        annotations: props.classification.annotations,
```

Comment:
Not a blocking comment for this PR, but requiring the classification here will break the subject viewer in Talk for any future projects that use data subjects.

Reply:
Should I use a default blank annotation, or wait for the Portal approach once React 16 is through?

Reply:
Or do you mean regarding #4484?

Reply:
Sorry, I mean that Talk doesn't have workflows or classifications. Neither do collections and favourites, which also display subjects. The subject viewer can't display classifications. Only the components in `/app/classifier/annotation-renderer` can render annotations.

`classification` and `annotation` should be optional props, with sensible defaults. With #4484 `classification.annotations` will always be an empty array while you're classifying.

Reply:
okay, so I'll provide those defaults in `file-viewer/index.jsx` for the time being, and then clean the tree of references to workflow/classification etc... once I can move canvas-viewer to a portal?

Reply:
I'm not sure that portals would fix this, but React 16 is still a way off anyway. Just be aware that file viewer components will be rendered outside the classifier so any classifier props that are passed down must be optional.